src/main/java/com/flightstats/hub/cluster/DistributedAsynchronousLockRunner.java

```diff
@@ -1,98 +1,62 @@
 package com.flightstats.hub.cluster;
 
 import com.google.inject.Inject;
-import org.apache.curator.framework.CuratorFramework;
-import org.apache.curator.framework.recipes.locks.InterProcessSemaphoreMutex;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import lombok.extern.slf4j.Slf4j;
 
+import java.util.Optional;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 
 /**
  * DistributedAsynchronousLockRunner is intended for long or short running processes.
  */
+@Slf4j
 public class DistributedAsynchronousLockRunner {
-    private final static Logger logger = LoggerFactory.getLogger(DistributedAsynchronousLockRunner.class);
-
-    private final CuratorFramework curator;
-    private final ExecutorService singleThreadExecutor;
-    private final Leadership leadership;
+    private final DistributedLeadershipLockManager leadershipLockManager;
```

Comment:
Its a long name. Not sure if we can creatively shorten this. But is should communicate its intent clearly which it is now :)

Reply:
can we just say Leader instead of leadership !

Reply:
Yeah... I definitely didn't know what a `CuratorLock` was off-hand and it wasn't clear that it was non-blocking. I couldn't think of something better and I tend to err on the side of super long names in those cases.

Reply:
Yeah, I can do that! And whoops, I thought the name you were referring to was the LockRunner.